File: app/src/main/java/com/example/lorddarthart/weathertest/DayofWeekConversion.java
package com.example.lorddarthart.weathertest;

public class DayofWeekConversion {
    public String convert(String code){
        switch (code){
            case  "Mon":{
                return "Monday";
            }
            case  "Tue":{
                return "Tuesday";
            }
            case  "Wed":{
                return "Wednesday";
            }
            case  "Thu":{
                return "Thusday";
            }
            case  "Fri":{
                return "Friday";
            }
            case  "Sat":{
                return "Saturday";
            }
            case  "Sun":{
                return "Sunday";
            }
        }
        return "";
    }
}

File: app/src/main/java/tk/lorddarthart/weathertest/util/converter/DayofWeekConversion.kt
package tk.lorddarthart.weathertest.util.converter

import tk.lorddarthart.weathertest.util.network.translator.YandexTranslate.translateToLocale

object DayofWeekConversion {
    fun convert(code: String): String {
        when (code) {
            "Mon" -> {
                return translateToLocale("Monday")
            }
            "Tue" -> {
                return translateToLocale("Tuesday")
            }
            "Wed" -> {
                return translateToLocale("Wednesday")
            }
            "Thu" -> {
                return translateToLocale("Thursday")
            }
            "Fri" -> {
                return translateToLocale("Friday")
            }
            "Sat" -> {
                return translateToLocale("Saturday")
            }
            "Sun" -> {
                return translateToLocale("Sunday")
            }
            else -> {
                return translateToLocale("Error")
            }
        }
    }
}
